### PR TITLE
[SDPA-2727] Added support for custom modules

### PR DESCRIPTION
**Pull request by tim-yao**
**JIRA issue:** https://digital-engagement.atlassian.net/browse/SDPA-2727

### Changed

1.  Added "modules" config in tide config, so we can register modules in custom site.

### CODE REVIEW

File: jest.config.js [hunk 1]
```diff
@@ -15,6 +15,7 @@ module.exports = {
     'node_modules/(?!(@dpc-sdp*|ol|ol-ext|storybook-addon-vue-info|@storybook*)/)'
   ],
   moduleNameMapper: {
+    '~(.*)$': '<rootDir>/examples/vic-gov-au/$1', // Add nuxt root alias in jest
```
Comment (tim-yao) [inline jest.config.js]: This is for jest to recognise the nuxt root dir alias.

File: packages/ripple-nuxt-tide/lib/module.js [hunk 2]
```diff
@@ -127,6 +130,27 @@ const nuxtTide = function (moduleOptions) {
...
+        options.extendFilters.push(customModuleMappingFilters)
+      }
+      if (customModule.moduleHook) {
+        const customModuleModuleHook = require(`${appDir}${options.nuxtRootPath}/tide/modules/${name}/module`)
+        if (typeof customModuleModuleHook === 'function') {
+          customModuleModuleHook.call(this)
+        }
+      }
+    }
+  }
```
Comment (tim-yao) [inline packages/ripple-nuxt-tide/lib/module.js]: dynamic load the module files by the modules config.

File: examples/vic-gov-au/tide/tide.config.js [hunk 3]
```diff
@@ -11,6 +11,15 @@ const tideConfig = {
...
     // landingPageComponents: {}
+  },
+
+  modules: {
+    'event': {
+      moduleHook: true,
+      config: true,
+      mappingFilters: true,
+      pageTypes: true
+    }
```
Comment (tim-yao) [inline examples/vic-gov-au/tide/tide.config.js]: This is the proposal to let custom sites register their modules. I copied our core `event` modules to custom site for testing at this stage.

File: packages/components/Organisms/ImageGallery/no-ssr/ImageGallery.vue [hunk 4]
```diff
@@ -62,6 +62,7 @@
 
 <script>
 import breakpoint from '@dpc-sdp/ripple-global/mixins/breakpoint'
+import RplIcon from '@dpc-sdp/ripple-icon'
```
Comment (tim-yao) [inline packages/components/Organisms/ImageGallery/no-ssr/ImageGallery.vue]: This is for fixing an error of rpl-icon not registered.

File: packages/ripple-nuxt-tide/lib/core/config-loader.js [hunk 5]
```diff
@@ -0,0 +1,212 @@
+const path = require('path')
+const fs = require('fs')
+const appDir = require('app-root-path')
+const kebabCase = require('lodash.kebabcase')
+
+// Groups of our configs
+const configGroups = {
```
Comment (dylankelly) [inline packages/ripple-nuxt-tide/lib/core/config-loader.js]: Not critical, but i think we should rethink this. It's not clear what these are or what they are doing.

**Pull request merged**